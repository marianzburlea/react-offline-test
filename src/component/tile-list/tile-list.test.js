import React from 'react'
import { create } from 'react-test-renderer'
import { render, unmountComponentAtNode } from 'react-dom'
import { act } from 'react-dom/test-utils'
import TileList from '.'
import fakeResponse from './tile-list.mock-data'

let container

beforeEach(() => {
  container = document.createElement('div')
  document.body.appendChild(container)
})

afterEach(() => {
  unmountComponentAtNode(container)
  container.remove()
  container = null
})

describe('@tile list', () => {
  it('should match the snapshot', () => {
    const tileList = create(<TileList />)
    expect(tileList.toJSON()).toMatchSnapshot()
  })

  it('should show a list of energy generation', async () => {
    jest
      .spyOn(window, 'fetch')
      .mockImplementation(() => {
        const fetchResponse = {
          json: () => Promise.resolve(fakeResponse),
        }

        return Promise.resolve(fetchResponse)
      })

    await act(async () => {
      render(<TileList />, container)
    })

    expect(container.textContent).toBe('biomass4.8coal2.5imports8.7gas46.5nuclear16.1other0.3hydro0.9solar14.6wind5.6biomasscoalimportsgasnuclearotherhydrosolarwind015304560perc')

    window.fetch.mockRestore()
  })
})
