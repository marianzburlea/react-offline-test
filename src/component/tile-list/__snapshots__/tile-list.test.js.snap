// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@tile list should match the snapshot 1`] = `
<div
  className="sc-gqjmRU jGsDzS"
>
  0
</div>
`;
